x-pack/legacy/plugins/siem/public/pages/detection_engine/components/signals/helpers.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get, isEmpty } from 'lodash/fp';
+import { esKuery } from '../../../../../../../../../src/plugins/data/common';
+import { esFilters } from '../../../../../../../../../src/plugins/data/public';
+import {
+  DataProvider,
+  DataProvidersAnd,
+} from '../../../../components/timeline/data_providers/data_provider';
+import { Ecs } from '../../../../graphql/types';
+
+interface FindValueToChangeInQuery {
+  field: string;
+  valueToChange: string;
+}
+
+const templateFields = [
+  'host.name',
+  'host.hostname',
+  'host.domain',
+  'host.id',
+  'host.ip',
+  'client.ip',
+  'destination.ip',
+  'server.ip',
+  'source.ip',
+  'network.community_id',
+  'user.name',
+  'process.name',
+];
+
+export const findValueToChangeInQuery = (
+  keuryNode: esKuery.KueryNode,
+  valueToChange: FindValueToChangeInQuery[] = []
+): FindValueToChangeInQuery[] => {
+  let localValueToChange = valueToChange;
+  if (keuryNode.function === 'is' && templateFields.includes(keuryNode.arguments[0].value)) {
+    localValueToChange = [
+      ...localValueToChange,
+      {
+        field: keuryNode.arguments[0].value,
+        valueToChange: keuryNode.arguments[1].value,
+      },
+    ];
+  }
+  return keuryNode.arguments.reduce(
+    (addValueToChange: FindValueToChangeInQuery[], ast: esKuery.KueryNode) => {
+      if (ast.function === 'is' && templateFields.includes(ast.arguments[0].value)) {
+        return [
+          ...addValueToChange,
+          {
+            field: ast.arguments[0].value,
+            valueToChange: ast.arguments[1].value,
+          },
+        ];
+      }
+      if (ast.arguments) {
+        return findValueToChangeInQuery(ast, addValueToChange);
+      }
+      return addValueToChange;
+    },
+    localValueToChange
+  );
+};
+
+export const replaceTemplateFieldFromQuery = (query: string, ecsData: Ecs) => {
+  if (query !== '') {
```

Comment:
Do you want to a trim first to ensure empty spaces don't count? `if(query.trim() !== '')`